### PR TITLE
Dnd remove default column value

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
This PR removes the default values of the two columns added in the migration numbers 30 and 31.
The default values were removed as adding a new column with default values to a table with a large number of rows can cause MySQL to lock the table for writing while it inserts the default value for each row.

#### Migration Performance Timing

On Oracle MySQL 5.7.12 with the following data size-

1. `focalboard_categories` - 1.32 million
2. `focalboard_category_boards` - 1.21 million

The migrations took-

1. `add_category_sort_order` took 16.2 seconds
4. `add_category_boards_sort_order` took 10.8575 seconds

Raw migration logs -

<details>
  <summary>Raw migration logs-</summary>
  
  ```
{"timestamp":"2022-12-14 17:43:09.916 +05:30","level":"debug","msg":"== Applying all remaining migrations ====================","caller":"app/plugin_api.go:970","plugin_id":"focalboard","current_version":"29"}
{"timestamp":"2022-12-14 17:43:09.918 +05:30","level":"info","msg":"2022/12/14 17:43:09 ==  add_category_sort_order: migrating  ============================================================\n","caller":"io/io.go:429","plugin_id":"focalboard","source":"plugin_stderr"}
{"timestamp":"2022-12-14 17:43:26.123 +05:30","level":"info","msg":"2022/12/14 17:43:26 ==  add_category_sort_order: migrated (16.2044s)  ==================================================\n2022/12/14 17:43:26 ==  add_category_boards_sort_order: migrating  =====================================================\n","caller":"io/io.go:429","plugin_id":"focalboard","source":"plugin_stderr"}
{"timestamp":"2022-12-14 17:43:36.981 +05:30","level":"info","msg":"2022/12/14 17:43:36 ==  add_category_boards_sort_order: migrated (10.8575s)  ===========================================\n2022/12/14 17:43:36 ==  move_boards_category_to_end: migrating  ========================================================\n","caller":"io/io.go:429","plugin_id":"focalboard","source":"plugin_stderr"}
{"timestamp":"2022-12-14 17:43:38.904 +05:30","level":"info","msg":"2022/12/14 17:43:38 ==  move_boards_category_to_end: migrated (1.9226s)  ===============================================\n","caller":"io/io.go:429","plugin_id":"focalboard","source":"plugin_stderr"}
```
</details>

<details>
  <summary>Raw database count query output-</summary>

  ```
  mysql> select count(id) from focalboard_categories;
  +-----------+
  | count(id) |
  +-----------+
  |   1327104 |
  +-----------+
  1 row in set (0.28 sec)
  
  mysql> select count(id) from focalboard_category_boards;
  +-----------+
  | count(id) |
  +-----------+
  |   1219584 |
  +-----------+
  1 row in set (0.28 sec)
  ```
</details>

### CODE REVIEW

**Comment by harshilsharma63** [comment]
@sbishel @ogi-m please create a fresh bug bash server once this PR gets merged into `main` to let the updated migrations run.

File: server/services/store/sqlstore/category.go [hunk 1]
```diff
@@ -172,6 +172,8 @@ func (s *SQLStore) categoriesFromRows(rows *sql.Rows) ([]model.Category, error)
 
 	for rows.Next() {
 		category := model.Category{}
+		var nullableSortOrder *int
```
Comment (mgdelacroix) [inline server/services/store/sqlstore/category.go:175]: Although the approach works fine, I'd suggest to change it for consistency and do the same thing we've done in other tables, which is to create a `categoryFields` method to return a `[]string` with all the fields and use it in all methods that later call `categoriesFromRows`

In those fields you can use `COALESCE` to set the default value, so the scan code can reliably expect a value set for the nullable column

Comment (wiggin77) [reply server/services/store/sqlstore/category.go:175]: I agree - this would be consistent with the other store methods.

Comment (harshilsharma63) [reply server/services/store/sqlstore/category.go]: Thanks for pointing this out. Done.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

**Comment by sbishel** [comment]
@harshilsharma63 - Please resolve conflicts. Thanks.

**Comment by sbishel** [comment]
@wiggin77 - Could you double check the fixes for merge conflicts?

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:   No merge conflicts.

**Review by sbishel — APPROVED** [review]

**Comment by sbishel** [comment]
@harshilsharma63 - I don't think this got cherry picked to `release-7.7`, could you take a look? Do we need it for `v7.7.0`?

**Pull request merged**